src/main/java/org/dependencytrack/event/kafka/processor/VulnerabilityScanResultProcessor.java

```diff
@@ -151,10 +151,12 @@ private void addVulnerability(final PersistenceManager pm, final UUID componentU
             query.setFilter("uuid == :uuid");
             query.setParameters(componentUuid);
             Component component = query.executeUnique();
-            component.addVulnerability(vuln);
-            component = pm.makePersistent(component);
-            var finding = new FindingAttribution(component, vuln, getAnalyzerIdentity(scanner), null, null);
-            pm.makePersistent(finding);
+            if(!contains(vuln, component)) {
+                component.addVulnerability(vuln);
+                component = pm.makePersistent(component);
+                var finding = new FindingAttribution(component, vuln, getAnalyzerIdentity(scanner), null, null);
+                pm.makePersistent(finding);
+            }
             trx.commit();
```

Comment:
The `commit` can also be moved into the `if` block. If we don't modify anything, there's no point in committing the transaction. It will be rolled back in the `finally` block.